### PR TITLE
[LM-10244] Upgrade SDK to Android 13

### PR DESCRIPTION
**Pull request by sidcr7-likeminds**
## Description

**Features**

- 

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [ ] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [ ] Label added
- [ ] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [ ] Description about the Feature/ Bugs added

### CODE REVIEW

File: chatmm/src/main/java/com/likeminds/chatmm/chatroom/detail/view/ChatroomDetailFragment.kt [hunk 1]
```diff
@@ -3324,6 +3413,46 @@ class ChatroomDetailFragment :
...
+                    ViewUtils.showShortToast(requireContext(), response.errorMessage)
+                }
+
+                is ChatroomDetailViewModel.ErrorMessageEvent.SetChatroomTopic -> {
+                    ViewUtils.showShortToast(requireContext(), response.errorMessage)
+                }
+
+                is ChatroomDetailViewModel.ErrorMessageEvent.SubmitPoll -> {
+                    ViewUtils.showShortToast(requireContext(), response.errorMessage)
+                }
```
Comment (ishaan1607) [inline chatmm/src/main/java/com/likeminds/chatmm/chatroom/detail/view/ChatroomDetailFragment.kt:3455]: these are repeated errors, please check

File: chatmm/src/main/java/com/likeminds/chatmm/chatroom/detail/view/ChatroomDetailFragment.kt [hunk 2]
```diff
@@ -3372,20 +3501,23 @@ class ChatroomDetailFragment :
...
+                        context?.getString(R.string.lm_chat_sorry_unfortunately_we_could_not_submit_your_choices_please_try_again)
                     )
                 }
 
                 is ChatroomDetailViewModel.ErrorMessageEvent.EditChatroomTitle -> {
                     ViewUtils.showLongSnack(binding.root, response.errorMessage)
                 }
+
+                is ChatroomDetailViewModel.ErrorMessageEvent.BlockMember -> TODO()
+                is ChatroomDetailViewModel.ErrorMessageEvent.SendDMRequest -> TODO()
```
Comment (ishaan1607) [inline chatmm/src/main/java/com/likeminds/chatmm/chatroom/detail/view/ChatroomDetailFragment.kt:3520]: repeated code

File: chatmm/src/main/java/com/likeminds/chatmm/media/customviews/CanvasView.kt [hunk 3]
```diff
@@ -272,19 +271,19 @@ internal class CanvasView : View, OnRotationGestureListener {
...
+//        val textLength = paintForMeasureText.measureText(text)
+//        val lengthOfChar = textLength / text.length.toFloat()
+//        val restWidth = canvas.width - textX // text-align : right
+//        val numChars =
+//            if (lengthOfChar <= 0) 1 else floor((restWidth / lengthOfChar).toDouble())
+//                .toInt() // The number of characters at 1 line
+//        val modNumChars = if (numChars < 1) 1 else numChars
+//        var y = textY
+//        var i = 0
+//        val len = text.length
```
Comment (ishaan1607) [inline chatmm/src/main/java/com/likeminds/chatmm/media/customviews/CanvasView.kt:286]: why commented?

File: chatmm/src/main/java/com/likeminds/chatmm/utils/ExtrasUtil.kt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+object ExtrasUtil {
+
+    //returns the extras passed with if-else check
+    @Suppress("DEPRECATION")
+    fun <T> getParcelable(argument: Bundle?, argumentName: String, clazz: Class<T>): T? {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            argument?.getParcelable(argumentName, clazz)
+        } else {
+            argument?.getParcelable(argumentName)
+        }
```
Comment (ishaan1607) [inline chatmm/src/main/java/com/likeminds/chatmm/utils/ExtrasUtil.kt:15]: use other method, this will lead to a crash

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: chatmm/src/main/java/com/likeminds/chatmm/media/customviews/CanvasView.kt (hunk 3 above)
Comment (sidcr7-likeminds) [reply chatmm/src/main/java/com/likeminds/chatmm/media/customviews/CanvasView.kt:286]: as this code is not being used. removing the code from the code base

**Review by ishaan1607 — APPROVED** [review]

**Pull request merged**